runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -154,7 +154,7 @@ public class CorfuRuntime {
      * File watcher for SSL key store to support auto hot-swapping.
      */
     @Getter
-    private Optional<FileWatcher> sslCertWatcher;
+    private Optional<FileWatcher> sslCertWatcher = Optional.empty();
```

Comment:
Just wondering, why isn't this being made volatile? It looks like different threads can invoke the CorfuRuntime methods that interact with this.

Reply:
Although I think the setting of this particular variable in the `connect` method is fine because it's protected by a `synchronize` block, I think there is some inconsistency with respect to the threading model. For example, `connect` is thread safe, but not `shutdown`.

Reply:
One thing to mention is that corfu runtime doesn't support stop and resume (once it's shut down it stays in stopped mode), so the risk of unsafe `shutdown` seems small.

Reply:
Thanks @zfrenette. I have made sslCertWatcher `volatile` as a precaution.

Reply:
Expecting concurrency, if `volatile` is not enough, what else is needed?

Right now it gets closed like this - 
`sslCertWatcher.ifPresent(FileWatcher::close);`